tests/test_text_classification.py

```diff
@@ -1 +1,132 @@
+
+from src.aiai_eval.exceptions import (
+    InvalidEvaluation,
+    MissingLabel,
+    WrongFeatureColumnName,
+)
+from src.aiai_eval.task_configs import SENT
+from src.aiai_eval.text_classification import TextClassification
+
+
+@pytest.fixture(scope="module")
+def dataset():
+    yield load_dataset("DDSC/angry-tweets", split="train")
+
+
+@pytest.fixture(scope="module")
+def text_clf(evaluation_config):
+    yield TextClassification(task_config=SENT, evaluation_config=evaluation_config)
+
+
+@pytest.fixture(scope="module")
+def tokenizer():
+    yield AutoTokenizer.from_pretrained("pin/senda")
+
+
+class TestPreprocessDataTransformer:
+    @pytest.fixture(scope="class")
+    def preprocessed(self, dataset, text_clf, tokenizer):
+        yield text_clf._preprocess_data_transformer(
+            dataset=dataset,
+            framework="pytorch",
+            tokenizer=tokenizer,
+        )
+
+    def test_spacy_framework_throws_exception(self, dataset, text_clf, tokenizer):
+        with pytest.raises(InvalidEvaluation):
+            text_clf._preprocess_data_transformer(
+                dataset=dataset,
+                framework="spacy",
+                tokenizer=tokenizer,
+            )
+
+    def test_preprocessed_is_dataset(self, preprocessed):
+        assert isinstance(preprocessed, Dataset)
+
+    def test_preprocessed_columns(self, preprocessed):
+        assert list(preprocessed.features.keys()) == [
+            "label",
+            "input_ids",
+            "token_type_ids",
+            "attention_mask",
+        ]
+
+    def test_throw_exception_if_feature_column_name_is_wrong(
+        self, dataset, evaluation_config, tokenizer
+    ):
+        # Create copy of the sentiment analysis task config, with a wrong feature
+        # column name
+        sent_cfg_copy = deepcopy(SENT)
+        sent_cfg_copy.feature_column_name = "wrong_name"
+
+        # Create a text classification task with the wrong feature column name
+        text_clf_copy = TextClassification(
+            task_config=sent_cfg_copy, evaluation_config=evaluation_config
+        )
+
+        # Attempt to preprocess the dataset with the wrong feature column name
+        with pytest.raises(WrongFeatureColumnName):
+            text_clf_copy._preprocess_data_transformer(
+                dataset=dataset,
+                framework="pytorch",
+                tokenizer=tokenizer,
+            )
+
+
+class TestPreprocessDataPyTorch:
+    @pytest.fixture(scope="class")
+    def preprocessed(self, dataset, text_clf, tokenizer):
+        yield text_clf._preprocess_data_pytorch(
+            dataset=dataset,
+            tokenizer=tokenizer,
+        )
+
+    def test_preprocessed_is_list(self, preprocessed):
+        assert isinstance(preprocessed, list)
+
+
+class TestCreateNumericalLabels:
+    @pytest.fixture(scope="class")
+    def examples(self):
+        yield dict(label=["POSITIVE", "POSITIVE", "NEGATIVE", "NEUTRAL"])
```

Comment:
"POSITIVE" is here twice? Is this a mistake or am i just not understanding the point?

Reply:
It's just a silly example, intended to be there twice :)